src/networking/torii.rs

```diff
@@ -1,45 +1,65 @@
 use crate::{
     cache::{mst_cache::MSTCache, pending_tx_cache::PendingTxCache},
+    client::query::Query,
     consensus::sumeragi::Sumeragi,
-    model::{block::Blockchain, commands::isi::Command, tx::Transaction},
+    model::tx::Transaction,
 };
 use std::{
     io::prelude::*,
     net::TcpListener,
-    thread::sleep,
     time::{Duration, Instant},
 };
 
+const TORII_URL: &str = "127.0.0.1:1337";
+const QUERY_REQUEST_HEADER: &[u8; 16] = b"GET / HTTP/1.1\r\n";
+const COMMAND_REQUEST_HEADER: &[u8; 25] = b"POST /commands HTTP/1.1\r\n";
+const _PUT: &[u8; 16] = b"PUT / HTTP/1.1\r\n";
+const OK: &[u8; 19] = b"HTTP/1.1 200 OK\r\n\r\n";
+
 #[allow(dead_code)]
 pub struct Torii {
     mst_cache: MSTCache,
     pending_tx_cache: PendingTxCache,
     consensus: Sumeragi,
     last_round_time: Instant,
-    blockchain: Blockchain,
 }
 
 impl Torii {
-    pub fn new() -> Self {
+    pub fn new(consensus: Sumeragi) -> Self {
         Torii {
             mst_cache: MSTCache::default(),
             pending_tx_cache: PendingTxCache::default(),
-            consensus: Sumeragi {},
+            consensus,
             last_round_time: Instant::now(),
-            blockchain: Blockchain::new(),
         }
     }
 
     pub fn start(&mut self) {
-        let listener = TcpListener::bind("127.0.0.1:1337").expect("could not start server");
+        let listener = TcpListener::bind(TORII_URL).expect("could not start server");
         for connection in listener.incoming() {
             match connection {
                 Ok(mut stream) => {
-                    let mut command_payload = Vec::new();
                     stream
-                        .read_exact(&mut command_payload)
-                        .expect("Command read failed.");
-                    self.receive(command_payload.into());
+                        .set_read_timeout(Some(Duration::new(2, 0)))
+                        .expect("Failed to set read timeout");
+                    stream
+                        .set_write_timeout(Some(Duration::new(2, 0)))
+                        .expect("Failed to set read timeout");
+                    let mut buffer = [0; 512];
+                    let _read_size = stream.read(&mut buffer).expect("Request read failed.");
+                    println!("Request: {}", String::from_utf8_lossy(&buffer[..]));
+                    if buffer.starts_with(COMMAND_REQUEST_HEADER) {
```

Comment:
I am not sure how many command or queries would torii need to handle, but we should at least discuss if the use of some web framework like rocket or actix is appropriate here. Without them code may become quite ugly in the future.

Reply:
There will be about 20 queries I think. But even then, I would do this in normal rust and not use a library. It can be easily read if we split up into files semantically (e.g., accounts, assets, ...). If we add another lib then we have to worry about updating it and also people reading the code have to suddenly go read some docs for a couple of hours to understand what is going on. Having this in normal rust is the best and we can easily control it and maintain it.

Reply:
We don't need to parse every command type here. We just use body to extract `Command` with payload as `Vec`.